### PR TITLE
fix: wrong link fixed for Architecture Overview - Package Architecture

### PR DESCRIPTION
**Pull request by mikeloguay**
## Relative link fixed pointing to Architecture Overview - Package Architecture

Just a minor fix on the URL

### CODE REVIEW

File: docs/backend-system/architecture/01-index.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Just like plugins, modules also have access to services and can depend on their
 
 A detailed explanation of the package architecture can be found in the
 [Backstage Architecture
-Overview](../../overview/architecture-overview.md#package-architecture). The
+Overview](../../overview/architecture-overview/#package-architecture). The
```
Comment (awanlin) [inline docs/backend-system/architecture/01-index.md:60]: The link is correct, it should be the `.md` file but it's the line break above that is causing this issue. Not sure why but Docusaurus does not like line breaks in links :(

**Review by awanlin — CHANGES_REQUESTED** [review]

File: docs/backend-system/architecture/01-index.md (hunk 1 above)
Comment (mikeloguay) [reply docs/backend-system/architecture/01-index.md]: Thanks @awanlin, you are completely right. I assumed the ".md" was the issue without even test it, sorry. I just tested locally and it is the new line the issue, as you said. It is now fixed.

Comment (awanlin) [reply docs/backend-system/architecture/01-index.md]: No worries, it's an odd issue so nothing to stress about. 👍

**Review by awanlin — APPROVED** [review]
Thanks again for this correction @mikeloguay, looks good barring any issues caught by the builds 🚀

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**